test/integration/typescript/pages/api/edge.ts

```diff
@@ -0,0 +1,14 @@
+import { NextRequest, RequestCookies } from 'next/server'
+
+export const config = {
+  runtime: 'experimental-edge',
+}
+
+const AsyncApiEndpoint = async (req: NextRequest) => {
+  // Test cookies type
+  const cookies: RequestCookies = req.cookies
+
+  return new Response(cookies instanceof RequestCookies ? 'ok' : 'error')
```

Comment:
This was failing for me locally, what could be the reason?

The following worked fine:

```ts
new RequestCookies(req.headers) instanceof RequestCookies // true
```

But when checking for `req.cookies` directly it does not:

```ts
req.cookies instanceof RequestCookies // false
```